src/sentry/workflow_engine/endpoints/project_detector_index.py

```diff
@@ -0,0 +1,71 @@
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from sentry.api.bases import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.issues import grouptype
+from sentry.workflow_engine.models import Detector
+
+
+class ProjectDetectorIndexEndpoint(ProjectEndpoint):
+    # TODO: We probably need a specific permission for detectors. Possibly specific detectors have different perms
+    # too?
+    permission_classes = (ProjectAlertRulePermission,)
+
+    def _get_validator(self, request, project, group_type_slug):
+        detector_type = grouptype.registry.get_by_slug(group_type_slug)
+        if detector_type is None:
+            raise ValidationError({"groupType": ["Unknown group type"]})
+
+        if detector_type.detector_validator is None:
+            raise ValidationError({"groupType": ["Group type not compatible with detectors"]})
+
+        return detector_type.detector_validator(
+            context={
+                "project": project,
+                "organization": project.organization,
+                "request": request,
+                "access": request.access,
+            },
+            data=request.data,
+        )
+
+    def get(self, request, project):
+        """
+        List a Project's Detectors
+        `````````````````````````
+        Return a list of detectors for a given project.
+        """
+        queryset = Detector.objects.filter(
+            organization_id=project.organization_id,
```

Comment:
Yeah, I'd move the relocation scope to project and remove org related things. Generally it's a pain to have both an org id and a project id, because when we transfer projects between orgs we often mess up changing the org id as well.